go-fuzz/testee.go

```diff
@@ -184,8 +190,7 @@ retry:
 		// deadlock we periodically read out stdout.
 		// This goroutine also collects crash output.
 		ticker := time.NewTicker(time.Second)
-		const N = 1 << 20
-		data := make([]byte, N)
+		data := buffer
```

Comment:
Eeee
this buffer is returned from TestBinary.test and then leaked to crasherQueue
we can't just do this
we can do something like this, but we can reuse the buffer only on periodic testee restart, on crashes we can either handoff the buffer to caller (assuming crashes are not so frequent, or if they are perf does not matter) or make a copy of the used part of the buffer and then reuse the original buffer

Reply:
I apologize, I don't follow. A few lines down, around line 215, I see:

```go
		trimmed := make([]byte, filled)
		copy(trimmed, data)
		t.outputC <- trimmed
```

So it seems like `data`/`buffer` doesn't leak. What am I missing?

Reply:
You are right!
I assumed we send the buffer to the caller as-is. I guess I did it to not pin the large buffer... perhaps...